sdk/communication/communication-chat/test/chatClient.spec.ts

```diff
@@ -22,47 +24,241 @@ describe("ChatClient", function() {
+      chatThreadClient.sendMessage(message).then((result) => {
+        chatThreadClient.updateMessage(result.id, {
+          content: "new content"
+        });
+      });
+    }).timeout(8000);
+
+    it("successfully listens to chatMessageReceivedEvents", function(done) {
+      function listener() {
+        done();
+      }
+
+      chatClient.on("chatMessageReceived", listener);
+
+      // Send message
+      const message = { content: `content` };
+      chatThreadClient.sendMessage(message);
+    }).timeout(8000);
+
+    it("successfully listens to chatMessageDeletedEvents", function(done) {
+      function listener() {
+        done();
+      }
+
+      chatClient.on("chatMessageDeleted", listener);
+
+      // Delete message
+      const message = { content: `content` };
+      chatThreadClient.sendMessage(message).then((result) => {
+        chatThreadClient.deleteMessage(result.id);
+      });
+    }).timeout(8000);
+
+    it("successfully listens to chatThreadCreatedEvents", function(done) {
+      function listener() {
+        done();
+      }
+
+      chatClient.on("chatThreadCreated", listener);
+
+      // Create thread
+      const request = {
+        topic: "test create thread event",
+        participants: [{ id: testUser }]
+      };
+      chatClient.createChatThread(request);
+    }).timeout(8000);
+
+    it("successfully listens to chatThreadDeletedEvents", function(done) {
+      function listener() {
+        done();
+      }
+
+      chatClient.on("chatThreadDeleted", listener);
+
+      // Delete thread
+      const request = {
+        topic: "test delete thread event",
+        participants: [{ id: testUser }]
+      };
+      chatClient.createChatThread(request).then((result) => {
+        chatClient.deleteChatThread(result.chatThread?.id!);
+      });
+    }).timeout(8000);
+
+    it("successfully listens to chatThreadPropertiesUpdatedEvents", function(done) {
+      function listener() {
+        done();
+      }
+
+      chatClient.on("chatThreadPropertiesUpdated", listener);
+
+      // Update thread
+      chatThreadClient.updateTopic("updated topic");
+    }).timeout(8000);
+
+    it("successfully listens to participantsAddedEvents", function(done) {
+      function listener() {
+        done();
+      }
+
+      chatClient.on("participantsAdded", listener);
+
+      // Add participant
+      const request = {
+        participants: [{ id: testUser2 }]
+      };
+      chatThreadClient.addParticipants(request);
+    }).timeout(8000);
+
+    it("successfully listens to participantsRemovedEvents", function(done) {
+      function listener() {
+        done();
+      }
+
+      chatClient.on("participantsRemoved", listener);
+
+      // Remove participant
+      chatThreadClient.removeParticipant(testUser2);
+    }).timeout(8000);
+
+    it("successfully listens to readReceiptReceivedEvents", function(done) {
+      // TODO: Read receipt notification is timing out even with increased timeout
```

Comment:
This pattern with the done handler is pretty cool. Let's see how flaky these tests are and if we might need to make them more robust in the future.
But the tests will actually give us a good idea how reliable our realtime events are 👍